### PR TITLE
Form for adding notes

### PR DESCRIPTION
**Pull request by dglebovskiy**
- Implemented a form for adding notes
- Added a component for adding notes
- Added new methods to notesService
- Added new states to state controller
- Updated the webapp layout

### CODE REVIEW

**Comment by evgeny-sav** [comment]
+1

File: notes_app/client/app/notes/notes_service.js [hunk 1]
```diff
@@ -4,9 +4,17 @@
...
-      'getList': function() {
-        return $http.get('http://localhost:8000/api/v1/note/');
+      'list': function() {
+        return $http.get(api_path);
+      },
+      'read': function(id) {
+        return $http.get(api_path + id + '/');
+      },
+      'create': function(note) {
+        note['owner'] = '/api/v1/user/1/';
```
Comment (alekseyC64) [inline notes_app/client/app/notes/notes_service.js]: magic line. Not all users have id=1

**Comment by evgeny-sav** [comment]
+1

File: notes_app/client/notes.js [hunk 2]
```diff
@@ -3,25 +3,34 @@
...
+      'abstract': true,
+      'templateUrl': 'templates/notes.tpl.html'
+    }).state({
+      'name': 'note.list',
+      'url': '',
+      'template': '<notes-list notes="$resolve.notes"></notes-list>',
+      'resolve': {
+        notes: function(notesService) {
+          return notesService.list().then(function(response) {
+            return response.data.objects;
```
Comment (alekseyC64) [inline notes_app/client/notes.js]: it looks like service code - work with request.

File: notes_app/client/notes.js [hunk 3]
```diff
@@ -3,25 +3,34 @@
...
+          }).catch(function(response) {
+            console.error('Error when loading notes');
```
Comment (alekseyC64) [inline notes_app/client/notes.js]: please move `.catch` to the service layer

File: notes_app/notes/api.py [hunk 4]
```diff
@@ -112,3 +112,7 @@ class Meta:
         list_allowed_methods = ['get', 'post']
         detail_allowed_methods = ['get', 'put', 'patch']
         authentication = BasicAuthentication()
+
+    def hydrate(self, bundle):
+        bundle.obj.owner = bundle.request.user
```
Comment (alekseyC64) [inline notes_app/notes/api.py]: assuming this code, does test of notes still work?

File: notes_app/client/app/notes/notes_service.js [hunk 5]
```diff
@@ -1,12 +1,40 @@
...
+  notesService.$inject = ['$http', '$log'];
 
-  function notesService($http) {
+  function notesService($http, $log) {
+    var api_path = 'http://localhost:8000/api/v1/note/';
     return {
-      'getList': function() {
-        return $http.get('http://localhost:8000/api/v1/note/');
+      'list': function() {
+        return new Promise(function(fulfill, reject) {
```
Comment (alekseyC64) [inline notes_app/client/app/notes/notes_service.js]: Angular has it's own implementation of promises - https://docs.angularjs.org/api/ng/service/$q

File: notes_app/client/app/users/user_service.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+(function() {
+  'use strict';
+
+  userService.$inject = ['$http', '$log'];
+  function userService($http, $log) {
+    var api_path = 'http://localhost:8000/api/v1/user/';
+    return {
+      'list': function() {
+        return new Promise(function(fulfill, reject) {
```
Comment (alekseyC64) [inline notes_app/client/app/users/user_service.js]: the same as above

File: notes_app/client/app/notes/note_add_form.component.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  'use strict';
+
+  NoteAddFormCtrl.$inject = ['notesService', 'userService'];
+  function NoteAddFormCtrl(notesService, userService) {
+    var self = this;
+    this.note = {};
+    this.users = [];
+
+    userService.list().then(function(response) {
+      self.users = response.objects;
```
Comment (alekseyC64) [inline notes_app/client/app/notes/note_add_form.component.js]: I think it should be re-implemented considering last service changes

File: notes_app/client/app/notes/notes_service.js [hunk 8]
```diff
@@ -1,12 +1,38 @@
...
+    var api_path = 'http://localhost:8000/api/v1/note/';
     return {
-      'getList': function() {
-        return $http.get('http://localhost:8000/api/v1/note/');
+      'list': function() {
+        return $http.get(api_path).then(function successHandler(response) {
+          return response.data;
+        }).catch(function errorHandler(response) {
+          $log.error('Problem with getting the user list from server')
+          return {};
```
Comment (ghost) [inline notes_app/client/app/notes/notes_service.js]: `list` shouldn't return an object, I think. Maybe empty array?

File: notes_app/client/app/notes/notes_service.js [hunk 9]
```diff
@@ -1,19 +1,38 @@
...
       },
       'read': function(id) {
-        return $http.get(api_path + id + '/');
+        return $http.get(api_path + id + '/').then(
+          function successHandler(response) {
+            return response.data;
+          }
+        ).catch(function errorHandler(response) {
+          $log.error('Problem with getting the user from server');
+          return {};
```
Comment (ghost) [inline notes_app/client/app/notes/notes_service.js]: I think here empty object is confusing. Code will wait an object, and if there is no object - it's better to return null, what do you think?

File: notes_app/client/app/notes/notes_service.js [hunk 10]
```diff
@@ -1,19 +1,38 @@
...
+          return response.data;
+        }).catch(function errorHandler(response) {
+          $log.error('Problem with getting the user list from server')
+          return {};
+        });
       },
       'read': function(id) {
-        return $http.get(api_path + id + '/');
+        return $http.get(api_path + id + '/').then(
+          function successHandler(response) {
```
Comment (ghost) [inline notes_app/client/app/notes/notes_service.js:19]: function name doesn't required in `.then` and `.catch`

**Review by ghost — APPROVED** [review]

**Pull request merged**